src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -366,7 +403,13 @@ public UserDetails call() throws AuthenticationException {
                 }
             });
             if (cacheMiss[0] != null) {
-                cacheMiss[0].updateUserInfo();
+                threadPoolExecutor.execute(new Runnable() {
+                    @Override
+                    public void run() {
+                        cacheMiss[0].updateUserInfo();
```

Comment:
for diagnostic purposes I would recommend switching the executing thread's name to include the name of the user that resolution is being performed on... that way if a specific user is blocking or taking excessively long then an admin will see this from repeated thread dumps. 

Also I would capture the time taken to do the update and log at say INFO if it takes longer than 1s or WARNING if longer than 5 or 10s (perhaps make the time limits configurable via system properties... no need to expose in UI IMHO... but others may see differently)